### PR TITLE
Add support to run benchmark against distribution url

### LINKED ISSUE #4073: Add option to run benchmark against OS artifact url
The nightly benchmarks currently depend on the distribution manifest file to get the location of the OpenSearch distribution tarball and then set up the cluster and run benchmark. 
There have been a few requests to run ad-hoc benchmark tests using nightly benchmark framework against the distribution tarball provided by requester. Since the tarball was created locally and had no manifest associated with it we could not accept the request and guided the user to run it locally. 
Add a feature in the workflow to be able to run the benchmark workflow using either a distribution manifest url or the actual artifact url.

### PR DESCRIPTION
**Pull request by rishabh6788**
### Description
Add support to run benchmark against distribution url.  See https://github.com/opensearch-project/opensearch-build-libraries/pull/326 

### Issues Resolved
#4073 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/opensearch/benchmark-test.jenkinsfile [hunk 1]
```diff
@@ -209,21 +219,29 @@ pipeline {
             agent { label AGENT_LABEL }
             steps {
                 script {
-                    if (BUNDLE_MANIFEST_URL == '') {
+                    if (BUNDLE_MANIFEST_URL == '' && DISTRIBUTION_URL == '') {
```
Comment (peterzhuamazon) [inline jenkins/opensearch/benchmark-test.jenkinsfile:222]: Add DISTRIBUTION_VERSION too?

File: jenkins/opensearch/benchmark-test.jenkinsfile [hunk 2]
```diff
@@ -209,21 +219,29 @@ pipeline {
...
                         currentBuild.result = 'ABORTED'
-                        error("Performance Tests failed to start. Missing parameter: BUNDLE_MANIFEST_URL.")
+                        error("Benchmark Tests failed to start. Provide either BUNDLE_MANIFEST_URL or DISTRIBUTION_URL with DISTRIBUTION_VERSION")
                     }
-                    def bundleManifestObj = downloadBuildManifest(
+                    if (BUNDLE_MANIFEST_URL != '') {
+                        def bundleManifestObj = downloadBuildManifest(
                         url: BUNDLE_MANIFEST_URL,
                         path: BUNDLE_MANIFEST
                     )
```
Comment (peterzhuamazon) [inline jenkins/opensearch/benchmark-test.jenkinsfile:230]: Can you fix the spacing and brackets alignment here? Rather confusing.

File: jenkins/opensearch/benchmark-test.jenkinsfile [hunk 3]
```diff
@@ -241,16 +259,23 @@ pipeline {
                     }
                     steps {
                         script {
-                            def bundleManifestObj = downloadBuildManifest(
+                            if (BUNDLE_MANIFEST_URL != '') {
+                                def bundleManifestObj = downloadBuildManifest(
                                 url: BUNDLE_MANIFEST_URL,
                                 path: BUNDLE_MANIFEST
                             )
```
Comment (peterzhuamazon) [inline jenkins/opensearch/benchmark-test.jenkinsfile:266]: Same as above.

**Review by peterzhuamazon — COMMENTED** [review]
Hi @rishabh6788 ,

Add some comments.

Thanks.

File: jenkins/opensearch/benchmark-test.jenkinsfile (hunk 1 above)
Comment (rishabh6788) [reply jenkins/opensearch/benchmark-test.jenkinsfile:222]: Not mandatory in this context. 
We need `DISTRIBUTION_URL` or `BUNDLE_MANIFEST_URL` and the DISTRIBUTION_VERSION exception is anyway caught by the python code. 
Safe to pass for now.

File: jenkins/opensearch/benchmark-test.jenkinsfile (hunk 3 above)
Comment (rishabh6788) [reply jenkins/opensearch/benchmark-test.jenkinsfile]: Fixed!

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**